datasets/arxiv_dataset/arxiv_dataset.py

```diff
@@ -0,0 +1,138 @@
+"""
+
+_HOMEPAGE = "https://www.kaggle.com/Cornell-University/arxiv"
+_LICENSE = "https://creativecommons.org/publicdomain/zero/1.0/"
+
+_ID = "id"
+_SUBMITTER = "submitter"
+_AUTHORS = "authors"
+_TITLE = "title"
+_COMMENTS = "comments"
+_JOURNAL_REF = "journal-ref"
+_DOI = "doi"
+_REPORT_NO = "report-no"
+_CATEGORIES = "categories"
+_LICENSE = "license"
+_ABSTRACT = "abstract"
+_VERSIONS = "versions"
+_UPDATE_DATE = "update_date"
+_AUTHORS_PARSED = "authors_parsed"
+
+_FILENAME = "arxiv-metadata-oai-snapshot.json"
+
+
+class ArxivDataset(datasets.GeneratorBasedBuilder):
+    """arXiv Dataset: arXiv dataset and metadata of 1.7M+ scholarly papers across STEM"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    @property
+    def manual_download_instructions(self):
+        return """\
+    You need to go to https://www.kaggle.com/Cornell-University/arxiv,
+    and manually download the dataset. Once it is completed,
+    a zip folder named archive.zip will be appeared in your Downloads folder
+    or whichever folder your browser chooses to save files to. Extract that folder
+    and you would get a arxiv-metadata-oai-snapshot.json file
+    You can then move that file under <path/to/folder>.
+    The <path/to/folder> can e.g. be "~/manual_data".
+    arxiv_dataset can then be loaded using the following command `datasets.load_dataset("arxiv_dataset", data_dir="<path/to/folder>")`.
+    """
+
+    def _info(self):
+        feature_names = [
+            _ID,
+            _SUBMITTER,
+            _AUTHORS,
+            _TITLE,
+            _COMMENTS,
+            _JOURNAL_REF,
+            _DOI,
+            _REPORT_NO,
+            _CATEGORIES,
+            _LICENSE,
+            _ABSTRACT,
+            _VERSIONS,
+            _UPDATE_DATE,
+            _AUTHORS_PARSED,
+        ]
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features({k: datasets.Value("string") for k in feature_names}),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        path_to_manual_file = os.path.join(os.path.abspath(os.path.expanduser(dl_manager.manual_dir)), _FILENAME)
+        if not os.path.exists(path_to_manual_file):
+            raise FileNotFoundError(
+                "{} does not exist. Make sure you insert a manual dir via `datasets.load_dataset('arxiv_dataset', data_dir=...)` that includes a file name {}. Manual download instructions: {})".format(
+                    path_to_manual_file, _FILENAME, self.manual_download_instructions
+                )
+            )
+        return [datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"path": path_to_manual_file})]
+
+    def _generate_examples(self, path=None, title_set=None):
+        """ Yields examples. """
+
+        data = []
+        for line in open(path, "r"):
```

Comment:
it is complaining about not having  `encoding="utf-8"` defined change line to -> `for line in open(path,  encoding="utf-8"):`

Reply:
btw why are you loading json line by line. Look at the example here -> https://github.com/huggingface/datasets/blob/master/templates/new_dataset_script.py#L162

Reply:
@cceyda I have fixed the code and still it is showing this error
![Screenshot (21)](https://user-images.githubusercontent.com/33005287/101454729-7272db80-3957-11eb-8e81-d68246397764.png)